contract/models/contract.py

```diff
@@ -452,16 +452,27 @@ def recurring_create_invoice(self):
         This method triggers the creation of the next invoices of the contracts
         even if their next invoicing date is in the future.
         """
-        return self._recurring_create_invoice()
+        invoice = self._recurring_create_invoice()
+        if invoice:
+            self.message_post(
+                body=_(
+                    'Contract manually invoiced: '
```

Comment:
This can be triggered manually or by the cron, so the text `Contract manually invoiced:` is not correct.

Reply:
it's not triggered with the cron

Reply:
the cron call `_recurring_create_invoice`

Reply:
Could it be triggered by other (possibly custom) code?

Reply:
> Could it be triggered by other (possibly custom) code?

no, this meant to be used in the UI

Reply:
OK, the similar name distracted me.

Reply:
Since the method is small, I'd add a new method for manual creation, this will proved a cleaner API.